File: src/my_project/easy_problems/from201to250/longer_contiguous_segments_of_ones_and_zeroes.py
from typing import List 

class Solution:
    def checkZeroOnes(self, s: str) -> bool:
        
        count_ones = 0
        count_zeroes = 0
        max_ones = 0
        max_zeroes = 0
        len_s: int = len(s)
        answer: bool = False
        
        
        if len_s == 1:
            if s == "1":
                answer = True 
                return answer 
            else: 
                return answer 
            
        s = s + "a"
        len_s += 1
        
        for i in range(len(s)-1):
            
            if s[i] =="1":
                count_ones +=1 
                if s[i+1] == "1":
                    continue
                else:
                    if count_ones > max_ones:
                        max_ones = count_ones
                        count_ones = 0
                    else: count_ones = 0          
            elif s[i] == "0": 
                count_zeroes +=1 
                
                if s[i+1] == "0":
                    continue
                else:
                    if count_zeroes > max_zeroes:
                        max_zeroes = count_zeroes
                        count_zeroes = 0
                    else: 
                        count_zeroes = 0
              
        if max_ones > max_zeroes:
            answer = True     
        
        return answer 
    
solution = Solution()

print(solution.checkZeroOnes(s = "1101"))

print(solution.checkZeroOnes( s = "111000"))

print(solution.checkZeroOnes(s = "110100010"))

File: my_project_typescript/easy_problems_typescript/from101to150/longerContiguousSegmentsOfOnesAndZeroes.ts
function checkZeroOnes(s: string): boolean {
    let countOnes: number = 0
    let countZeroes: number = 0
    let maxOnes: number = 0
    let maxZeroes: number = 0
    let lenS: number = s.length
    let answer: boolean = false 

    if (lenS === 1){
        if (s === "1"){
            answer = true 
            return answer 
        }
        else {
            return answer 
        }
    }

    s = s + 'a'
    lenS += 1

    for (let i = 0; i < lenS; i++){
        if (s[i] === "1"){
            countOnes += 1
            if (s[i+1] === "1"){
                continue
            }
            else{
                if (countOnes > maxOnes){
                    maxOnes = countOnes
                    countOnes = 0
                }
                else {
                    countOnes = 0
                }
            }
        }
        else if (s[i] === "0"){
            countZeroes += 1
            if (s[i+1] === "0"){
                continue
            }
            else {
                if (countZeroes > maxZeroes){
                    maxZeroes = countZeroes
                    countZeroes = 0
                }
                else {
                    countZeroes = 0
                }
            }
        }
    }

    if (maxOnes > maxZeroes){
        answer = true 
    }
    return answer 
};

console.log(checkZeroOnes("1101"))

console.log(checkZeroOnes("111000"))

console.log(checkZeroOnes("110100010"))

